IPv6 related NPE in ServiceInfoImpl.getHostAddresses()

jmdns version: 3.5.6. 

I've got a few automated crash reports with this stack trace, with `getHostAddress()` called in a `ServiceListener.serviceResolved()` callback:

```
Caused by: java.lang.NullPointerException
at javax.jmdns.impl.ServiceInfoImpl.getHostAddresses(ServiceInfoImpl.java:420)
at javax.jmdns.impl.ServiceInfoImpl.getHostAddress(ServiceInfoImpl.java:404)
```

looking at `getHostAddresses()`:

```
 @Override
    public String[] getHostAddresses() {
        Inet4Address[] ip4Aaddresses = this.getInet4Addresses();
        Inet6Address[] ip6Aaddresses = this.getInet6Addresses();
        String[] names = new String[ip4Aaddresses.length + ip6Aaddresses.length];
        for (int i = 0; i < ip4Aaddresses.length; i++) {
            names[i] = ip4Aaddresses[i].getHostAddress();
        }
        for (int i = 0; i < ip6Aaddresses.length; i++) {
            names[i + ip4Aaddresses.length] = "[" + ip6Aaddresses[i].getHostAddress() + "]";    
        }
        return names;
    }
```

line 420 is:

```
names[i + ip4Aaddresses.length] = "[" + ip6Aaddresses[i].getHostAddress() + "]";    
```

The only possible thing `null` on that line is `ip6Aaddresses[i]`. 
So I suppose `getInet6Addresses()` can return an `Inet6Address[]` where some elements are `null`.

I have not checked if v3.5.8 would fix this issue but I doubt it. 
Anyway, since my program does not make use of ipv6 at all, I worked-around this issue issue calling `ServiceInfoImpl.getInet4Address().getHostAddress()`.
Still it would be nice to get the bottom line of that NPE.

--- a/src/main/java/javax/jmdns/impl/ServiceInfoImpl.java
+++ b/src/main/java/javax/jmdns/impl/ServiceInfoImpl.java
@@ -418,16 +418,40 @@ public class ServiceInfoImpl extends ServiceInfo implements DNSListener, DNSStat
      */
     @Override
     public String[] getHostAddresses() {
-        Inet4Address[] ip4Aaddresses = this.getInet4Addresses();
-        Inet6Address[] ip6Aaddresses = this.getInet6Addresses();
-        String[] names = new String[ip4Aaddresses.length + ip6Aaddresses.length];
-        for (int i = 0; i < ip4Aaddresses.length; i++) {
-            names[i] = ip4Aaddresses[i].getHostAddress();
+        Inet4Address[] ip4Addresses = this.getInet4Addresses();
+        Inet6Address[] ip6Addresses = this.getInet6Addresses();
+
+        // Default to empty arrays if null
+        if (ip4Addresses == null) {
+            ip4Addresses = new Inet4Address[0];
         }
-        for (int i = 0; i < ip6Aaddresses.length; i++) {
-            names[i + ip4Aaddresses.length] = "[" + ip6Aaddresses[i].getHostAddress() + "]";
+        if (ip6Addresses == null) {
+            ip6Addresses = new Inet6Address[0];
         }
-        return names;
+
+        // Use ArrayList to store valid addresses
+        List<String> addresses = new ArrayList<>();
+
+        // Add non-null IPv4 addresses
+        for (Inet4Address ip4 : ip4Addresses) {
+            if (ip4 != null) {
+                addresses.add(ip4.getHostAddress());
+            } else {
+                logger.info("IPv4 address is null");
+            }
+        }
+
+        // Add non-null IPv6 addresses (with brackets)
+        for (Inet6Address ip6 : ip6Addresses) {
+            if (ip6 != null) {
+                addresses.add("[" + ip6.getHostAddress() + "]");
+            } else {
+                logger.info("IPv6 address is null");
+            }
+        }
+
+        // Convert the ArrayList to an array and return
+        return addresses.toArray(new String[0]);
     }
 
     /**

